src/acbm/assigning/plots.py

```diff
@@ -229,7 +230,7 @@ def plot_desire_lines(
     bin_size: int,
     boundaries: gpd.GeoDataFrame,
     sample_size: Optional[int] = None,
-    save_dir: Optional[str] = None,
+    save_dir: str | None | Path = None,
```

Comment:
How do I use pathlib for this now? Do we need to edit the scripts? Currently, this is how we pass the save_dir: [link](https://github.com/Urban-Analytics-Technology-Platform/acbm/blob/96fe6e1dcf5b8b77515c03d48c0857af20e00afa/scripts/3.3_assign_facility_all.py#L349)

Reply:
I think it's since we added `acbm.root_path` (as this is a `Path`) and when this combines with a `str` (with the `/`) it also returns a `Path`.

Reply:
Yeah good point on the other scripts. I updated the other `save_dir` in `plots.py` [ce523a8](https://github.com/Urban-Analytics-Technology-Platform/acbm/pull/41/commits/ce523a8d4f22510a41c0fde2ab4c00272571e8db) since also called with `Path` arg in the scripts. It might be worth adding a type checker at some point for this to the pre-commit/actions too (e.g. mypy/pyright) - I'll open an issue.